### PR TITLE
Working example for Atlas-encryptionAtRest-roles with a single tf apply

### PR DESCRIPTION
**Pull request by zohar-mongo**
## Description
Working example for Atlas-encryptionAtRest-roles with a single `terraform apply`. This is a workaround that overcomes the cycle issue. The existing example has a bug where the `iam_assumed_role_arn` argument is populated with the `var.aws_iam_role_arn` variable, which has a default of an empty string. 

The `output "aws_iam_role_arn"` was removed since it's not in use, and it creates confusion.

Link to any related issue(s):

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the Terraform contribution guidelines
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

### CODE REVIEW

**Comment by themantissa** [comment]
@shum @leofigy @coderGo93 can you all let me know your thoughts?  A nice PR from one of our CSAs!

File: examples/atlas-encryptionAtRest-roles/aws-roles.tf
Comment (leofigy) [inline examples/atlas-encryptionAtRest-roles/aws-roles.tf]: pretty cool !

**Comment by shum** [comment]
Not a TF expert so can't comment on whether this approach is "best practice™️"  , but this is a nifty hack to workaround the problem and looks good to me!

**Comment by themantissa** [comment]
@zohar-mongo thank you again for knocking this out.  Instead of changing the original example I think we should  consider spliting it into two.  Keep the original as is (though w/ the bug fix here for the unused variables fyi @coderGo93) but actually make it two directories, one called examples/atlas-encryptionAtRest-roles-two-step and examples/atlas-encryptionAtRest-roles-one-step-workaround.  Then if we get it to one step without a workaround we add that one.  Would you mind doing the rework on that?

**Comment by zohar-mongo** [comment]
@themantissa Sure

**Comment by zohar-mongo** [comment]
@themantissa Updated the pull request per your request :)

**Review by themantissa — APPROVED** [review]
Looks good, thank you for knocking this out.  There's one artifact (the second-step 2nd directory in the one step example) but we can fix that up post merge.  I'll have one of our devs take a look again and double check but this should help a lot for now.  Thanks!

**Comment by themantissa** [comment]
@coderGo93  or @leofigy can one of you give one more look?  I think we'll need to clean up one of the second-step directories (in the copy from the current example) for the one step workaround but we can do that later I think.

**Comment by zohar-mongo** [comment]
@themantissa Thanks, I didn't modify the "second step" folder. From what I understand, this is the next step for configuring the encryption at rest with the *role_id* we exported in the first step (creating the role_id with the correct iam_assumed_role_arn in one step was the problem)

**Comment by themantissa** [comment]
Ah, I follow.  We could move the second into one big example I believe @zohar-mongo but we can improve later.   Just waiting for one more review from those selected.  Thanks!

**Review by leofigy — APPROVED** [review]
LGTM :) , thanks

File: examples/atlas-encryptionAtRest-roles-two-step/versions.tf
Comment (leofigy) [inline examples/atlas-encryptionAtRest-roles-two-step/versions.tf]: just could you remove this comment from here

**Comment by themantissa** [comment]
@nikhil-mongo  I'm going to merge this soon but it could use some clean-up potentially.   Can you take a look and let me know if you have any concerns before I merge?

File: examples/atlas-encryptionAtRest-roles-one-step-workaround/aws-roles.tf
Comment (nikhil-mongo) [inline examples/atlas-encryptionAtRest-roles-one-step-workaround/aws-roles.tf]: Please remove this extra line.

File: examples/atlas-encryptionAtRest-roles-two-step/second_step/variables.tf
Comment (nikhil-mongo) [inline examples/atlas-encryptionAtRest-roles-two-step/second_step/variables.tf:3]: The default values passed as an empty string just add more lines and are of no use. @themantissa if you think we can keep it, its ok. Else please remove these as well.

File: examples/atlas-encryptionAtRest-roles-two-step/variables.tf
Comment (nikhil-mongo) [inline examples/atlas-encryptionAtRest-roles-two-step/variables.tf:3]: The default values passed as an empty string just add more lines and are of no use. @themantissa if you think we can keep it, its ok. Else please remove these as well.

**Review by nikhil-mongo — CHANGES_REQUESTED** [review]
Please look at the below changes and have them done.

**Review by nikhil-mongo — APPROVED** [review]
LGTM. @themantissa if you want those variables to be kept or not, I'm good with both. But this is good to be merged.

**Comment by themantissa** [comment]
@nikhil-mongo thank you.  We can keep for now and if you'd like to do a bit of clean up later it would be appreciated.   Thank you.

**Pull request merged**
